Build failure with fmt > 8.0

After upgrading fmt (8.0.1-2 -> 8.1.1-1) there is now a build error:

```
In file included from /usr/include/spdlog/fmt/fmt.h:25,
                 from /usr/include/spdlog/common.h:36,
                 from /usr/include/spdlog/spdlog.h:12,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/util/logger.h:35,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/common.h:36,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/util/timer.h:39,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.h:38,
                 from /build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.cc:32:
/usr/include/fmt/core.h: In instantiation of 'constexpr fmt::v8::detail::value<Context> fmt::v8::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; fmt::v8::detail::type <anonymous> = fmt::v8::detail::type::custom_type; T = ScanMode&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]':
/usr/include/fmt/core.h:1855:77:   required from 'constexpr fmt::v8::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {ScanMode&}; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {ScanMode}]'
/usr/include/fmt/core.h:1872:38:   required from 'constexpr fmt::v8::format_arg_store<Context, fmt::v8::remove_cvref_t<Args>...> fmt::v8::make_format_args(Args&& ...) [with Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {ScanMode&}]'
/usr/include/fmt/core.h:3119:44:   required from 'std::string fmt::v8::format(fmt::v8::format_string<T ...>, T&& ...) [with T = {ScanMode&}; std::string = std::__cxx11::basic_string<char>; fmt::v8::format_string<T ...> = fmt::v8::basic_format_string<char, ScanMode&>]'
/build/.cache/pikaur/build/gerbera-git/src/gerbera-git/src/content/autoscan.cc:124:5:   required from here
/usr/include/fmt/core.h:1728:7: error: static assertion failed: Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
 1728 |       formattable,
      |       ^~~~~~~~~~~
make[2]: *** [CMakeFiles/libgerbera.dir/build.make:230: CMakeFiles/libgerbera.dir/src/content/autoscan.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/libgerbera.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

```

--- a/src/util/logger.h
+++ b/src/util/logger.h
@@ -32,12 +32,27 @@
 #ifndef __LOGGER_H__
 #define __LOGGER_H__
 
+#include <fmt/format.h>
 #include <spdlog/spdlog.h>
+#include <type_traits>
 
 #define log_debug SPDLOG_DEBUG
 #define log_info SPDLOG_INFO
 #define log_warning SPDLOG_WARN
 #define log_error SPDLOG_ERROR
 #define log_js SPDLOG_INFO
+
+#if FMT_VERSION >= 80100
+template <typename T>
+struct fmt::formatter<T, std::enable_if_t<std::is_enum_v<T>, char>>
+    : formatter<std::underlying_type_t<T>> {
+    template <typename FormatContext>
+    auto format(const T& value, FormatContext& ctx) -> decltype(ctx.out())
+    {
+        return fmt::formatter<std::underlying_type_t<T>>::format(
+            static_cast<std::underlying_type_t<T>>(value), ctx);
+    }
+};
+#endif
 
 #endif // __LOGGER_H__

